// @contentful/app-scripts/lib/analytics.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.track = void 0;
const analytics_node_1 = __importDefault(require("analytics-node"));
// Public write key scoped to data source
const SEGMENT_WRITE_KEY = 'IzCq3j4dQlTAgLdMykRW9oBHQKUy1xMm';
/**
 *
 * @param {object} properties tracking properties
 * @param {string} properties.command triggered command e.g create-app-definition, upload, etc.
 * @param {boolean} properties.ci value if --ci flag has been set
 * @returns
 */
function track({ command, ci }) {
    if (process.env.DISABLE_ANALYTICS) {
        return;
    }
    try {
        const client = new analytics_node_1.default(SEGMENT_WRITE_KEY, {
            errorHandler: () => {
                // noop
            },
        });
        client.track({
            event: 'app-cli-app-scripts',
            properties: {
                command,
                ci: String(ci),
            },
            anonymousId: Date.now(),
            timestamp: new Date(),
        });
        // eslint-disable-next-line no-empty
    }
    catch (e) {
        // ignore any error, to not block the call
    }
}
exports.track = track;
